mtp_noms_ops/apps/settings/views.py

```diff
@@ -152,3 +152,38 @@ def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context['prisons'] = self.request.user_prisons
         return context
+
+
+class JobInformationView(FormView):
+    title = _('Help us improve this service')
+    template_name = 'settings/job-information.html'
+    success_url = reverse_lazy('security:dashboard')
+    form_class = JobInformationForm
+
+    def get_success_url(self):
+        if 'next' in self.request.GET:
+            return '{path}?{query}'.format(
+                path=self.success_url,
+                query=urlencode({'next': self.request.GET['next']})
+            )
+        return self.success_url
+
+    def form_valid(self, *args, **kwargs):
+        if has_provided_job_information(self.request.user):
```

Comment:
I'm not overly happy about this early return, but it handles and edge case of submitting the form if it happens to be open in 2 browsers and submitted twice accidentally (getting a HTTP 500 error from the API).

I couldn't see how the choose prison handled this however, although it doesn't seem to have the same issue. Suggestions are welcome.

Reply:
that's quite an unlikely edge case i think. also, i have no issues with early returns personally